fastparquet/dataframe.py

```diff
@@ -91,7 +90,7 @@ def cat(col):
     df = OrderedDict()
     for t, col in zip(types, cols):
         if str(t) == 'category':
-            df[str(col)] = Categorical([], categories=cat(col), fastpath=True)
+            df[str(col)] = Categorical.from_codes([], categories=cat(col))
```

Comment:
Will this work with all of the pandas versions that we support? At the moment, [it looks like 1.3+](https://github.com/dask/dask/blob/fd31b9b67e105f4c63b391f5c24959e39e27a0cc/continuous_integration/environment-3.8.yaml#L28).

Reply:
pandas test fails right now with this change.

Reply:
Test fails because of #859, the failure is unrelated to this change